charts/linkerd2/templates/trafficsplit-crd.yaml

```diff
@@ -14,16 +13,58 @@ metadata:
     {{.Values.global.controllerNamespaceLabel}}: {{.Values.global.namespace}}
 spec:
   group: split.smi-spec.io
-  version: v1alpha1
   scope: Namespaced
   names:
     kind: TrafficSplit
+    listKind: TrafficSplitList
     shortNames:
-      - ts
+    - ts
     plural: trafficsplits
     singular: trafficsplit
-  additionalPrinterColumns:
-  - name: Service
-    type: string
-    description: The apex service of this split.
-    JSONPath: .spec.service
+  version: v1alpha1
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+  preserveUnknownFields: false
+  validation:
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          required:
+            - service
+            - backends
+          properties:
+            service:
+              description: The apex service of this split.
+              type: string
+            matches:
+              description: The HTTP route groups that this traffic split should match.
+              type: array
+              items:
+                type: object
+                required: ['kind', 'name']
+                properties:
+                  kind:
+                    description: Kind of the matching group.
+                    type: string
+                    enum:
+                      - HTTPRouteGroup
+                  name:
+                    description: Name of the matching group.
+                    type: string
+            backends:
+              description: The backend services of this split.
+              type: array
+              items:
+                type: object
+                required: ['service', 'weight']
+                properties:
+                  service:
+                    description: Name of the Kubernetes service.
+                    type: string
+                  weight:
+                    description: Traffic weight value of this backend.
+                    type: number
```

Comment:
I think the failing test is because of this line.  In version v1alpha1, this is allowed to be a quantity string: https://github.com/servicemeshinterface/smi-spec/blob/master/apis/traffic-split/v1alpha1/traffic-split.md

Reply:
Thank you for pointing this out. I changed the weight to include `x-kubernetes-int-or-string`.

Initially I wanted to put this under the version itself and get rid of the top-level validation, e.g
```yaml
- name: v1alpha1
  schema:
    openAPIV3Schema:
    ...
```

however, this prompted error messages when doing `linkerd check --pre`; think resources with v1beta1 can't make use of the new versioned schema field. [Check this comment out for more info on this topic](https://github.com/kubernetes/kubernetes/issues/82443#issuecomment-529033688).